LanguageFeatures/Patterns/Exhaustiveness/exhaustiveness_sealed_A03_t01.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with a sealed class as a
+/// matched type are always exhaustive
+///
+/// @description Check that if a matched value of a switch statement is not a
+/// sealed class then flow analysis treats this statement as a not exhaustive
+/// @author sgrekhov22@gmail.com
+
+abstract class A {}
+class B extends A {}
+
+void main() {
+  int x;
+  A a = B();
+  switch (a) {
+    case B():
+      print("x not assigned here");
+    case _ :
+      x = 1;
+  }
+  print(x);
```

Comment:
I don't think line 24 is a very helpful statement in this test.

The type `A` is not always-exhaustive, so we'd expect the switch statement to be OK even though it isn't exhaustive ... but it _is_ exhaustive because it ends in a catch-all case. So we don't get any information about whether the compiler thought the switch statement was or wasn't required to be exhaustive.

Reply:
Ups, my bad. Class `B` here shouldn't extends `A`, of course. The main idea of this tests is to check that because `B`  is not a sealed class then it can be a hidden subtype of `A` and `B` somewhere in another library. Therefore in spite of the fact that `case B():` looks unreacable it isn't so and therefore this should be a compile-time error. Updated. Does it makes sense now?